benches/search_comparison.rs

```diff
@@ -0,0 +1,276 @@
+                &mut group,
+                i,
+                false,
+            );
+            construction_bench_case::<MAX, T, _>("btreeset", make_btreeset, &mut group, i, false);
+            construction_bench_case::<MAX, T, _>("ordsearch", make_this, &mut group, i, false);
+        }
+        group.finish();
+    }
+}
+
+fn search_bench_case<const MAX: usize, T, Coll>(
+    name: &str,
+    setup_fun: impl Fn(Vec<T>) -> Coll,
+    search_fun: impl Fn(&Coll, T) -> Option<&T>,
+    group: &mut BenchmarkGroup<WallTime>,
+    i: &usize,
+    duplicates: bool,
+) where
+    T: TryFrom<usize> + Ord + std::ops::Rem<Output = T> + num_traits::ops::wrapping::WrappingMul,
+    <T as TryFrom<usize>>::Error: core::fmt::Debug,
+{
+    group.bench_with_input(BenchmarkId::new(name, i), i, |b, i| {
+        let size = *i;
+        let v: Vec<T> = if duplicates {
+            (0..*i)
+                .map(|int| {
+                    let int = std::cmp::min(int, MAX);
+                    T::try_from(int / 16 * 16).unwrap()
+                })
+                .collect()
+        } else {
+            (0..*i)
+                .map(|int| {
+                    let int = std::cmp::min(int, MAX);
+                    T::try_from(int).unwrap()
+                })
+                .collect()
+        };
+        let mut r = 0usize;
+        let c = setup_fun(v);
+        b.iter(|| {
+            r = r.wrapping_mul(1664525).wrapping_add(1013904223);
+            let r = std::cmp::min(r, MAX);
+            let x = T::try_from(r % size).unwrap();
+            let _res = black_box(search_fun(&c, x));
+        })
+    });
+}
+
+fn construction_bench_case<const MAX: usize, T, Coll>(
+    name: &str,
+    setup_fun: impl Fn(Vec<T>) -> Coll,
+    group: &mut BenchmarkGroup<WallTime>,
+    i: &usize,
+    duplicates: bool,
+) where
+    T: TryFrom<usize>
+        + Ord
+        + std::ops::Rem<Output = T>
+        + num_traits::ops::wrapping::WrappingMul
+        + Clone,
+    <T as TryFrom<usize>>::Error: core::fmt::Debug,
+{
+    group.bench_with_input(BenchmarkId::new(name, i), i, |b, i| {
+        let size = *i;
+        let mut v: Vec<T> = if duplicates {
+            (0..*i)
+                .map(|int| {
+                    let int = std::cmp::min(int, MAX);
+                    T::try_from(int / 16 * 16).unwrap()
+                })
+                .collect()
+        } else {
+            (0..*i)
+                .map(|int| {
+                    let int = std::cmp::min(int, MAX);
+                    T::try_from(int).unwrap()
+                })
+                .collect()
+        };
+        let mut r = 0usize;
+        for e in v.iter_mut() {
+            r = r.wrapping_mul(1664525).wrapping_add(1013904223);
+            if duplicates {
+                let r = (r % size) / 16 * 16;
+                let r = std::cmp::min(r, MAX);
+                *e = T::try_from(r).unwrap();
+            } else {
+                let r = (r % size) * 2;
```

Comment:
You'll probably want to adjust this (and elsewhere?) to reflect the change in #9.